src/test/scala/org/broadinstitute/dsde/workbench/avram/db/TestComponent.scala

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import java.time.Instant
+
+import org.broadinstitute.dsde.workbench.avram.TestExecutionContext
+import org.broadinstitute.dsde.workbench.model.google.{GoogleProject, ServiceAccountKeyId}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Seconds, Span}
+import org.scalatest.Matchers
+import slick.dbio.DBIO
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.ExecutionContext
+
+trait TestComponent extends Matchers with ScalaFutures with AvramComponent {
+  override val profile: JdbcProfile = DbSingleton.ref.dataAccess.profile
+  implicit val executionContext: ExecutionContext = TestExecutionContext.testExecutionContext
+  implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(10, Seconds)))
+  val defaultServiceAccountKeyId = ServiceAccountKeyId("123")
+
+  def dbFutureValue[T](f: DataAccess => DBIO[T]): T = DbSingleton.ref.inTransaction(f).futureValue
```

Comment:
It would be nice to have something that gives back the `Future` to allow full use of the `FutureConcept` functions. Something like:
```
def db[T](f: DataAccess => DBIO[T]): Future[T] = DbSingleton.ref.inTransaction(f)
```
This would allow test code like (adapted from `CollectionComponentSpec`):
```
    val futureResult = db(_.collectionQuery.getCollectionByName(CommonTestData.collectionName).map(_.get))
    whenReady(futureResult) { collection =>
      collection.name shouldEqual CommonTestData.collectionName
      collection.samResource shouldEqual CommonTestData.samResource
      collection.createdBy shouldEqual CommonTestData.user1
    }
```
Could also have a `dbRun` or something that uses `futureValue` to execute the thing but throw away the result (for cases where we only care about the effect, like for `truncateAll()`). Just gives us a little more flexibility in our test code.

Reply:
I don't want to add this in the PR